src/components/FlavorStash/Note.js

```diff
@@ -0,0 +1,230 @@
+          </Button>
+        )}
+      </Fragment>
+    );
+  }
+
+  handleNoteEditor() {
+    this.setState({ editingNote: !this.state.editingNote });
+  }
+
+  handleNoteSubmit(values) {
+    const { actions } = this.props;
+    const { note, update } = values;
+
+    if (!note) {
+      actions.deleteNote(values);
+    } else if (update === true) {
+      actions.updateNote(values);
+    } else {
+      actions.createNote(values);
+    }
+    this.setState({ updatedNote: note || '' });
+
+    this.handleNoteEditor();
+  }
+
+  noteEditor() {
+    const { flavorId, note, userId } = this.props;
+    const { updatedNote } = this.state;
+    const update = Boolean(note?.[flavorId]?.note);
+
+    return (
+      <FinalForm
+        onSubmit={this.handleNoteSubmit}
+        initialValues={{
+          flavorId,
+          note: updatedNote || note?.[flavorId]?.note,
+          update,
+          userId
+        }}
+        render={({ handleSubmit, submitting, values }) => (
+          <Form noValidate onSubmit={handleSubmit}>
+            <Form.Row>
+              <Field name="note">
+                {({ input }) => (
+                  <Form.Group as={Col}>
+                    <InputGroup>
+                      <Form.Control
+                        {...input}
+                        as="textarea"
+                        placeholder="Flavor Note"
+                      />
+                    </InputGroup>
+                  </Form.Group>
+                )}
+              </Field>
+            </Form.Row>
+            {values?.note ? (
+              <Form.Row>
+                <h4>Preview:</h4>
+                <Col md="12">
+                  {
+                    /* eslint-disable-next-line no-sync */
+                    unified()
+                      .use(parse)
+                      .use(remark2react)
+                      .processSync(values.note).result
+                  }
+                </Col>
+              </Form.Row>
+            ) : null}
+            <Field name="userId">
+              {({ input }) => <Form.Control {...input} type="hidden" />}
+            </Field>
+            <Field name="flavorId">
+              {({ input }) => <Form.Control {...input} type="hidden" />}
+            </Field>
+            <Field name="update">
+              {({ input }) => <Form.Control {...input} type="hidden" />}
+            </Field>
+            <Form.Row>
+              <Form.Group>
+                <ButtonGroup>
+                  <Button
+                    className="button-animation"
+                    variant="primary"
+                    type="submit"
+                    disabled={submitting}
+                  >
+                    <FontAwesomeIcon icon="save" size="sm" />
+                    &nbsp;<span>Save</span>
+                  </Button>
+                  <Button
+                    onClick={e => this.handleNoteEditor(false, e)}
+                    className="button-animation button--cancel"
```

Comment:
This is likely something for a different pr, but I think this class name should be `button--animated`